MetaMask/metamask-mobile
# fix: finding and updating transaction errors in confirmation pages state

## **Description**
The PR fixes the issue - whenever loading balance is delayed, the Warning for needing more ETH is displayed and cannot perform the tx once the balance is loaded.
The reason for this issue was wrong state update for error, I fixed this but re-calculating error every time transaction / user balance / contracts balance changes. Thus as soon as balance is fetched the error goes away.

## **Manual testing steps**
1. Go to test-dapp
2. Simulate a slow response from the getBalance request
3. Click on Send EIP1559
4. See how warning for needing more ETH appears and Confirm button is disabled
5. Wait until correct balance is loaded
6. See how warning is removed and Confirm button is still enabled

## **Related issues**

_Fixes #6535

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
